core/corehttp/commands.go

```diff
@@ -117,11 +117,17 @@ func patchCORSVars(c *cmdsHttp.ServerConfig, addr net.Addr) {
 	c.SetAllowedOrigins(newOrigins...)
 }
 
-func commandsOption(cctx oldcmds.Context, command *cmds.Command) ServeOption {
+func commandsOption(cctx oldcmds.Context, command *cmds.Command, allowGet bool) ServeOption {
 	return func(n *core.IpfsNode, l net.Listener, mux *http.ServeMux) (*http.ServeMux, error) {
 
 		cfg := cmdsHttp.NewServerConfig()
-		cfg.SetAllowedMethods(http.MethodGet, http.MethodPost, http.MethodPut)
+		cfg.AllowGet = allowGet
+		corsAllowedMethods := []string{http.MethodPost, http.MethodPut}
```

Comment:
I'm not currently allowing PUT in the commands lib. Will that be an issue?

Reply:
AFAIK the only place where we use `PUT` is in "writable gateway",
which is on the Gateway port, but outside of `/api/v0` namespace.
Is there other use?

Reply:
I don't think so. I'll remove this.